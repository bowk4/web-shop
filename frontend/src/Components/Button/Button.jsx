import React, { useState } from "react";
import PropTypes from "prop-types";
import styles from "./Button.module.scss";

const Button = ({
  onClick,
  children,
  backgroundColor,
  hoverBackgroundColor,
  height,
  disabled,
  type,
  fontSize,
}) => {
  const [isHovered, setIsHovered] = useState(false);

  const buttonStyle = {
    backgroundColor:
      isHovered && !disabled ? hoverBackgroundColor : backgroundColor,
    height,
    cursor: disabled ? "default" : "pointer",
    fontSize,
  };

  return (
    <button
      type={type}
      style={buttonStyle}
      className={styles.button}
      disabled={disabled}
      onClick={onClick}
      onMouseEnter={() => setIsHovered(true)}
      onMouseLeave={() => setIsHovered(false)}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
  backgroundColor: PropTypes.string,
  hoverBackgroundColor: PropTypes.string,
  height: PropTypes.string,
  disabled: PropTypes.bool,
  fontSize: PropTypes.string,
};

Button.defaultProps = {
  type: "button",
  onClick: () => {},
  backgroundColor: "#905BFF",
  hoverBackgroundColor: "#a378ff",
  height: "40px",
  disabled: false,
  fontSize: "14px",
};

export default React.memo(Button);
