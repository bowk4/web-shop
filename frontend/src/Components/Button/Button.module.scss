@import "../../styles/abstracts/variables";
@import "../../styles/abstracts/mixins";

.button {
  @include flex-center;
  @include button;
  height: 40px;
  width: 100%;
  color: $ui-color-gray-white;
  border: none;

  &:disabled {
    background-color: $ui-color-gray-surface-1;
  }

  &:active {
    background-color: $ui-color-gray-selected-surface-2;
  }
}
