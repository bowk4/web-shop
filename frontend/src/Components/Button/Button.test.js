import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Button from "./Button";

describe("Button tests", () => {
  test("matches snapshot", () => {
    const { asFragment } = render(<Button>Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
