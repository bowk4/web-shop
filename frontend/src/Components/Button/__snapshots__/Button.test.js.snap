// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    style="background-color: rgb(144, 91, 255); height: 40px; cursor: pointer; font-size: 14px;"
    type="button"
  >
    Click me
  </button>
</DocumentFragment>
`;
